### PR TITLE
Clear external ids on older messages (part 1)

### PR DESCRIPTION
**Pull request by rowanseymour**
Depends on https://github.com/nyaruka/rapidpro/pull/870 which should be merged first

Part 2: create new partial index on non-null external ids
Part 3: remove `db_index=True` on `Msg.external_id`

### CODE REVIEW

File: temba/settings_common.py [hunk 1]
```diff
@@ -974,6 +974,10 @@
         'task': 'fail_old_messages',
         'schedule': crontab(hour=0, minute=0),
     },
+    "clear-old-msg-external-ids": {
+        'task': 'clear_old_msg_external_ids',
+        'schedule': crontab(hour=0, minute=0),
```
Comment (ericnewcomer) [inline temba/settings_common.py]: how bout hour 1? might as well stagger our dailies.

File: temba/msgs/tasks.py [hunk 2]
```diff
@@ -253,38 +241,40 @@ def handle_event_task():
...
+    Clears external_id on older messages to reduce the size of the index on that column. External ids aren't surfaced
+    anywhere and are only used for debugging channel issues, so are of limited usefulness on older messages.
+    """
+    threshold = timezone.now() - timedelta(days=30)  # 30 days ago
 
-    key = 'squash_systemlabels'
-    if not r.get(key):
-        with r.lock(key, timeout=900):
-            SystemLabel.squash_counts()
+    msg_ids = list(Msg.objects.filter(created_on__lt=threshold).exclude(external_id=None).values_list('id', flat=True))
```
Comment (ericnewcomer) [inline temba/msgs/tasks.py]: the first run of this is going to be one massive list of ids in memory

File: temba/settings_common.py (hunk 1 above)
Comment (rowanseymour) [inline temba/settings_common.py]: Updated

File: temba/msgs/tasks.py (hunk 2 above)
Comment (rowanseymour) [inline temba/msgs/tasks.py]: ~300,000,000 messages x 8 bytes = ~ 2.3GB

I think that's ok?

File: temba/msgs/tasks.py [hunk 3]
```diff
@@ -264,3 +264,19 @@ def purge_broadcasts_task():
 @nonoverlapping_task(track_started=True, name="squash_systemlabels")
 def squash_systemlabels():
     SystemLabel.squash_counts()
+
+
+@nonoverlapping_task(track_started=True, name='clear_old_msg_external_ids')
```
Comment (rowanseymour) [inline temba/msgs/tasks.py]: Note that this task doesn't have any timeout set on it - first run will take a looong time

**Pull request merged**